### PR TITLE
MONIT-30608: Update Proxy Retry to not retry if daily limit is reached

### PR DESCRIPTION
**Pull request by xuranchen**

### CODE REVIEW

**Review by amitw-vmware — APPROVED** [review]
LGTM

File: proxy/src/main/java/com/wavefront/agent/data/LogDataSubmissionTask.java [hunk 1]
```diff
@@ -65,6 +67,14 @@ Response doExecute() {
     return api.proxyLogs(AGENT_PREFIX + proxyId.toString(), logs);
   }
 
+  @Override
+  protected TaskResult handleStatus429() {
+    Metrics.newCounter(
+            new MetricName(entityType + "." + handle, "", "failed" + ".ingestion_limit_reached"))
+        .inc(this.weight());
+    return TaskResult.REMOVED;
```
Comment (laullon) [inline proxy/src/main/java/com/wavefront/agent/data/LogDataSubmissionTask.java:75]: Is ok to drop logs on a 429 ? Why not retry ?

**Review by iprotsiuk — APPROVED** [review]
tested with docker and local proxy, lgtm

File: proxy/src/main/java/com/wavefront/agent/data/LogDataSubmissionTask.java (hunk 1 above)
Comment (xuranchen) [reply proxy/src/main/java/com/wavefront/agent/data/LogDataSubmissionTask.java:75]: A 429 from VRLIC means that the daily ingestion limit has been reached

Comment (laullon) [reply proxy/src/main/java/com/wavefront/agent/data/LogDataSubmissionTask.java:75]: So, that's not coming from wavefront ?

Comment (xuranchen) [reply proxy/src/main/java/com/wavefront/agent/data/LogDataSubmissionTask.java:75]: Correct, its coming from the VRLIC backend

**Review by joannak-vmware — APPROVED** [review]

**Pull request merged**